app/src/vuex/modules/connection.js

```diff
@@ -76,16 +76,19 @@ export default function ({ apollo }) {
       }
       // HACK as polkadot addresses don't have a prefix
       if (isPolkadotAddress(address)) {
-        const selectedNetwork = await state.networks
+        let selectedNetwork = undefined
+        const substrateNetworks = await state.networks
           .filter(({ network_type }) => network_type === `polkadot`)
-          .find(
-            async (network) =>
-              await isValidPolkadotAddress(
-                address,
-                parseInt(network.address_prefix)
-              )[0]
-          )
-
+        for (let index = 0; index < substrateNetworks.length; index++) {
```

Comment:
why does the for loop work better than the filter?

Reply:
Filter and find are synchronous and doesn't work well with async code inside them. Same for forEach as I tested, but for loops works pretty well (https://stackoverflow.com/questions/43031988/javascript-efficiency-for-vs-foreach)